00_download_data/nbconverted/0_download_data.py

```diff
@@ -72,14 +73,39 @@
 
 md5_sum = get_ipython().getoutput('md5sum $me_filepath')
 print(md5_sum[0])
-assert md5_sum[0].split(' ')[0] == manifest_df.loc['methylation'].md5
+assert md5_sum[0].split(' ')[0] == manifest_df.loc['methylation_27k'].md5
 
 
-# ### Download RPPA data
+# ### Download 450k probe DNA methylation data (warning: large file, ~40GB)
+# 
+# This took me overnight (~12 hours) to download, although it could be faster on some connections...
 
 # In[7]:
 
 
+me_id, me_filename = manifest_df.loc['methylation_450k'].id, manifest_df.loc['methylation_450k'].filename
+url = 'http://api.gdc.cancer.gov/data/{}'.format(me_id)
+me_filepath = os.path.join(cfg.raw_data_dir, me_filename)
+
+if not os.path.exists(me_filepath):
+    urlretrieve(url, me_filepath)
```

Comment:
I also used `urlretrieve` in the past, but it failed to download recount2 files when they "moved" the URL. Seems like it doesn't support redirection. I had to change it and use the `requests` package ([here](https://github.com/miltondp/phenoplier/blob/641ba94c6618d3e98e5c5814fc50d61dc624ef39/libs/data/recount2.py#L103), in case you have the same problem in the future).

Reply:
This is good to know! I don't think these will move anytime soon, but I'll keep this in mind for the future.